api/api/validators/generate_teams_validator.py

```diff
@@ -7,23 +14,59 @@ def validate(self):
         self.validate_algorithm_type()
         self.validate_team_options()
         self.validate_algorithm_options()
-        self.validate_projects()
+        self.validate_algorithm_configs()
         self.validate_students()
 
     def _validate_schema(self):
-        pass
+        Schema(dict).validate(self.data)
+        Schema(list).validate(self.data.get("students"))
+        Schema(dict).validate(self.data.get("algorithm_options"))
+        Schema(dict).validate(self.data.get("team_generation_options"))
 
     def validate_algorithm_type(self):
-        pass
+        algorithm_type = self.data.get("algorithm_options").get("algorithm_type")
+        Schema(str).validate(algorithm_type)
+        try:
+            AlgorithmType(algorithm_type)
+        except ValueError:
+            raise SchemaError(
+                f"Algorithm type {algorithm_type} is not a valid algorithm type."
+            )
 
     def validate_algorithm_options(self):
-        pass
+        algorithm_option = dict.copy(self.data.get("algorithm_options"))
+        algorithm_type = AlgorithmType(algorithm_option.pop("algorithm_type"))
+        algorithm_option_cls = AlgorithmRunner.get_algorithm_option_class(
+            algorithm_type
+        )
+
+        algorithm_option_cls(**algorithm_option).validate()
+
+    def validate_algorithm_configs(self):
+        algorithm_type = AlgorithmType(
+            self.data.get("algorithm_options").get("algorithm_type")
+        )
+        algorithm_config = self.data.get("algorithm_config")
+        algorithm_config_cls = AlgorithmRunner.get_algorithm_config_class(
+            algorithm_type
+        )
 
-    def validate_projects(self):
-        pass
+        algorithm_config_cls(**algorithm_config).validate()
 
     def validate_students(self):
```

Comment:
Not really the kind of validation we need in this method.

Creating the `Student` objects is the job of the loader, the validator just needs to make sure the data is all good to go.

For example, something important to validate is that every student dict has a unique id, or that the ids in their `project_preference` list actually correlate to real projects that were passed in

Reply:
We can chat about this if this feels confusing, I don't think I properly communicated the semantic difference between classes accurately